wordpress-mobile/WordPress-Android
# Create site info block item

Fixes #13343 

I'm sorry for the size of this PR. I've tried extracting unrelated steps into separate PRs and not implementing parts of the logic but the rest seems really coupled together. Thinking back I probably could've left out the items interaction for another PR. What this PR does is that it introduces the `SiteInfoBlock`. That's the block that contains the icon, site name, URL and the `switchSite` button. 
- The icon is clickable and it opens a dialog the same way as it does in the current implementation. The dialog doesn't do anything yet and crashes the app on `TODO` items
- The title is clickable and opens a dialog to change the site title. The dialog interaction doesn't do anything yet and crashes the app on `TODO` items.
- The site URL opens the browser with the URL
- The switch site button opens the Switch Site activity (the result is not yet handled).

To test:
- Turn on the Improved My Site flag and restart the app
- Go to My Site fragment
- Notice the Site Info block is now visible
- Click on the URL 
- Notice the site URL opens in a browser
- Click on Switch site button 
- Notice the switch site activity is opened
- Click on the Title 
- Notice the title change dialog opens (dismiss will crash the app)
- Click on the site icon
- Notice the dialog opens (interaction will crash the app)

PR submission checklist:
